x-pack/plugins/apm/common/agent_configuration/runtime_types/duration_rt.ts

```diff
@@ -6,32 +6,68 @@
 
 import * as t from 'io-ts';
 import { either } from 'fp-ts/lib/Either';
-import { amountAndUnitToObject } from '../amount_and_unit';
+import moment, { unitOfTime } from 'moment';
+import { i18n } from '@kbn/i18n';
+import { amountAndUnitToObject, AmountAndUnit } from '../amount_and_unit';
+import { getRangeType } from './get_range_type';
 
-export const DURATION_UNITS = ['ms', 's', 'm'];
+function getDuration({ amount, unit }: AmountAndUnit) {
+  return moment.duration(amount, unit as unitOfTime.Base);
+}
+
+export function getDurationRt({
+  min,
+  max,
+  units
+}: {
+  min?: string;
+  max?: string;
+  units: string[];
+}) {
+  const minAmountAndUnit = min && amountAndUnitToObject(min);
+  const maxAmountAndUnit = max && amountAndUnitToObject(max);
```

Comment:
```suggestion
  const minDuration = amountAndUnitToMillis(min);
  const minDuration = amountAndUnitToMillis(max);
```

where `amountAndUnitToMillis` is:
```
function amountAndUnitToMillis(value: string) {
  const { amount, unit } = amountAndUnitToObject(value);
  return toMillis({ amount, unit });
}
```